tests/composite_pk/test_create.py

```diff
@@ -136,3 +137,12 @@ def test_update_or_create_user(self):
                 self.assertEqual(user.email, fields["defaults"]["email"])
                 self.assertEqual(user.email, f"user{user.id}@example.com")
                 self.assertEqual(count + 1, User.objects.count())
+
+    def test_save_default_pk_not_set(self):
+        with self.assertNumQueries(1):
+            Post().save()
+
+    def test_save_default_pk_set(self):
+        post = Post.objects.create()
+        with self.assertRaises(IntegrityError):
+            Post(tenant_id=1, id=post.id).save()
```

Comment:
```suggestion
            Post(tenant_id=post.tenant_id, id=post.id).save()
```
(nit) I think this is slightly more explicit